src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -199,7 +202,18 @@ private int AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePa
 
         private void AnalyzeDirectory(DirectoryInfo directoryPath, FileInfo configurationsFilePath, ReportFormat reportFormat, FileInfo outputFilePath, bool runTtk, bool verbose)
         {
-            var logger = CreateLogger(verbose);
+            // Check that output file path provided for sarif report
+            if (reportFormat == ReportFormat.Sarif && outputFilePath == null)
+            {
+                // We can't use the logger for this error,
+                // because we need to get the writer to create the logger, but this check has to be done before getting the writer:
+                Console.WriteLine("Output file path was not provided.");
+                return;
+            }
+
+            var reportWriter = this.GetReportWriter(reportFormat.ToString(), outputFilePath, directoryPath.FullName);
```

Comment:
Can you keep the `using` here?

```suggestion
            using var reportWriter = this.GetReportWriter(reportFormat.ToString(), outputFilePath, directoryPath.FullName);
```